main/tests/server/src/com/google/refine/expr/functions/strings/ToFromConversionTests.java

```diff
@@ -167,12 +170,22 @@ public void testToDate() throws CalendarParserException {
       // First string can be a locale identifier instead of a format string
       Assert.assertEquals(invoke("toDate", "2013-06-01","zh"), CalendarParser.parseAsOffsetDateTime("2013-06-01"));
       Assert.assertEquals(invoke("toDate", "01-六月-2013","zh","dd-MMM-yyyy"), CalendarParser.parseAsOffsetDateTime("2013-06-01"));
+      Assert.assertEquals(invoke("toDate", "01-六月-2013","zh", "MMM-dd-yyyy", "dd-MMM-yyyy"), CalendarParser.parseAsOffsetDateTime("2013-06-01"));
 
       // If a long, convert to string
       Assert.assertEquals(invoke("toDate", (long) 2012), invoke("toDate", "2012-01-01"));
 
       // If already a date, leave it alone
       Assert.assertEquals(invoke("toDate", CalendarParser.parseAsOffsetDateTime("2012-03-01")),CalendarParser.parseAsOffsetDateTime("2012-03-01"));
+
+      // FIXME: Date/times without timezone info should be interpreted as local time, not UTC
+//      Assert.assertEquals(invoke("toDate", "2013-06-01T13:12:11"), CalendarParser.parseAsOffsetDateTime("2013-06-01 13:12:11"));
```

Comment:
@wetneb I think this test should pass, but it fails due to UTC timezone offset code that you added. Can you review and see if you agree?

Reply:
@tfmorris I don't want to poke my eyes or mouth into this between both of you only to say this bit of history:
I recall Jacky in a chat asking me about formatter expectations (GREL, cell values, etc.) because he was indeed having issues with correct parsing behavior based on what we expected or not.  That probably should have been a signal for me to ask Antonin or someone else (who was still newish to our team), but I think that discussion did eventually happen between him and Antonin in the contributing PRs and original issues for Java Date that he worked on.  I recall Instant and Locale being 2 terms he asked me about concerning expectations and I didn't have enough expertise.

The ISO_INSTANT would be UTC, yes? https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/time/format/DateTimeFormatter.html#ISO_INSTANT
But the ISO_LOCAL_DATE_TIME would be local, no? https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/time/format/DateTimeFormatter.html#ISO_LOCAL_DATE_TIME

This is all I recall of the conversations with him, sorry.  Hope it helps!

Reply:
Speaking of which...which Resolver style do we feel is most useful for regular OpenRefine users? https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/time/format/ResolverStyle.html#STRICT

Reply:
@tfmorris that proposed test looks obviously sensible to me.

Broadly speaking, my user-side requirements for this would be:
- if I parse a date string and then extract some date part from it, the value I get should be consistent with what was in the original string
- the rendered version of the date should be consistent with the source string.

I have not checked but that might require changes to `datePart` and cell rendering, to make sure these work correctly for local date times.